app/exec/build-tasks-upload.ts

```diff
@@ -25,33 +27,55 @@ export class BuildTaskUpload extends cmdm.TfCommand {
     public exec(args: string[], options: cm.IOptions): any {
         var taskPath = args[0] || options['taskPath'];
         this.checkRequiredParameter(taskPath, 'taskPath', 'path to the task folder');
+        var deferred = Q.defer<agentifm.TaskDefinition>();
+        fs.readFile(path.join(taskPath, 'task.json'), 'utf8', (err, data) => {
+            if(err) {
+                return console.log(err);
+            }
 
-        var result: any = {};
-        result.task = { name: name };
-        var archive = archiver('zip');
-        archive.directory(taskPath, false);
-        //result.source = taskPath;
+            var taskJson = JSON.parse(data);
```

Comment:
This can throw.  Catch and reject with error that states invalid task json + err.message